tests/integration/test_fim/conftest.py

```diff
@@ -0,0 +1,131 @@
+import os
+from time import sleep
+import distro
+import pytest
+import re
+import subprocess
+import sys
+
+from typing import Any
+from pathlib import Path
+
+from wazuh_testing.constants.daemons import WAZUH_MANAGER
+from wazuh_testing.constants.paths.logs import WAZUH_LOG_PATH
+from wazuh_testing.constants.platforms import WINDOWS
+from wazuh_testing.modules.fim.patterns import MONITORING_PATH
+from wazuh_testing.tools.monitors.file_monitor import FileMonitor
+from wazuh_testing.tools.simulators.authd_simulator import AuthdSimulator
+from wazuh_testing.tools.simulators.remoted_simulator import RemotedSimulator
+from wazuh_testing.utils import file
+from wazuh_testing.utils.callbacks import generate_callback
+from wazuh_testing.utils.services import get_service
+
+
+@pytest.fixture()
+def file_to_monitor(test_metadata: dict) -> Any:
+    path = test_metadata.get('file_to_monitor')
+    path = os.path.abspath(path)
+
+    file.write_file(path)
+
+    yield path
+
+    file.remove_file(path)
+
+
+@pytest.fixture()
+def folder_to_monitor(test_metadata: dict) -> None:
+    path = test_metadata.get('folder_to_monitor')
+    path = os.path.abspath(path)
+
+    file.create_folder(path)
+
+    yield path
+
+    file.delete_path_recursively(path)
+
+
+@pytest.fixture()
+def fill_folder_to_monitor(test_metadata: dict) -> None:
+    path = test_metadata.get('folder_to_monitor')
+    amount = test_metadata.get('files_amount')
+    amount = 2 if not amount else amount
+
+    if not file.exists(path):
+        file.create_folder(path)
+
+    [file.write_file(Path(path, f'test{i}.log'), 'content') for i in range(amount)]
+        # file.write_file(Path(path, f'test{i}.log'), 'content')
+
+    yield
+
+    [file.remove_file(Path(path, f'test{i}.log')) for i in range(amount)]
+
+
+@pytest.fixture()
+def start_monitoring() -> None:
+    FileMonitor(WAZUH_LOG_PATH).start(generate_callback(MONITORING_PATH))
+
+
+@pytest.fixture(scope='module', autouse=True)
+def set_agent_config(request: pytest.FixtureRequest):
+    if not hasattr(request.module, 'test_configuration'):
+        return
+    if get_service() is WAZUH_MANAGER:
```

Comment:
We should think if we want this. I think these tests should only run on agents, not in manager.